### PR TITLE
Fix field order in send

### LINKED ISSUE #17: Send opcode broken
VM execution deviates from correct behavior when calling the `SEND` and `NBSEND` opcodes. Bug was detected running the go AVM against the cpp AVM.

### PR DESCRIPTION
**Pull request by hkalodner**
This fixes #17

### CODE REVIEW

File: vm/instructions.go [hunk 1]
```diff
@@ -992,29 +992,14 @@ func sendImpl(state *Machine) (value.TupleValue, value.Value, value.IntValue, va
...
-	}
-
-	val4, err := sendData.GetByInt64(3)
-	if err != nil {
-		return sendData, nil, value.NewInt64Value(0), value.NewInt64Value(0), value.NewInt64Value(0), mods, err
-	}
+	data, _ := sendData.GetByInt64(0)
+	val2, _ := sendData.GetByInt64(1)
+	val3, _ := sendData.GetByInt64(2)
+	val4, _ := sendData.GetByInt64(3)
```
Comment (ughe) [inline vm/instructions.go]: Why is this error no longer checked?

File: vm/instructions.go [hunk 2]
```diff
@@ -992,29 +992,14 @@ func sendImpl(state *Machine) (value.TupleValue, value.Value, value.IntValue, va
...
 
-	tokenType, ok2 := val2.(value.IntValue)
+	destination, ok2 := val2.(value.IntValue)
 	amount, ok3 := val3.(value.IntValue)
-	destination, ok4 := val4.(value.IntValue)
+	tokenType, ok4 := val4.(value.IntValue)
```
Comment (ughe) [inline vm/instructions.go]: Is there any documentation on where the ordering is: destination, amount, tokenType?

File: vm/instructions.go (hunk 1 above)
Comment (hkalodner) [reply vm/instructions.go]: We already do the size check before we hit these lines

Comment (ughe) [reply vm/instructions.go]: Size check right before makes sure there are 4 values:

https://github.com/OffchainLabs/arb-avm/blob/97575c895f23fd661b1eac1fdb5dc214ba52f97a/vm/instructions.go#L990-L993

File: vm/instructions.go (hunk 2 above)
Comment (hkalodner) [reply vm/instructions.go]: This had been incorrect in the spec (https://developer.offchainlabs.com/docs/AVM_Specification/), but is now updated.

Comment (ughe) [reply vm/instructions.go]: From the spec:

> Each Message is a 4-tuple [value, sender, amount, currency], where

So `destination` is the sender and `tokenType` is currency

**Pull request merged**